torch/_dynamo/eval_frame.py

```diff
@@ -884,6 +888,16 @@ def graph_with_interpreter(*args):
         graph,
     ).transform()
 
+    new_graph.meta["example_inputs"] = example_inputs
+    new_graph.meta["input_shape_constraints"] = constraints
```

Comment:
It looks like @ezyang would prefer keep `w_tensor` but get rid of `t_id` in the long run (There is 
a TODO). I verified that `weakref` can not be serialized however (doesn't work with `torch.save/load`). IIRC the exported Aten graph should be serializable right. Do we want to fix serialization of `weakref` or we want to keep `t_id` instead?

```
class Constraint:
    """
    This represents constraints on input tensor dimensions, e.g., requiring
    them to be fully polymorphic or within some range.  Don't create this
    class directly; instead, use :func:`torch._export.dynamic_dim`.
    """
    w_tensor: weakref.ReferenceType[torch.Tensor]
    # TODO: We don't need t_id; we can get it off of w_tensor
    t_id: int
    ```

Reply:
Tried a workaround by setting `w_tensor = None` which still still not make `Constraint` serializable. @tugsbayasgalan if you need it to be saved/loaded in your PR, maybe we can store `Constraint` as key-value pairs w/o `w_tensor` field.

Reply:
I don't need it to be serializable. But there are some other people who rely on the fact that torchdynamo.export result is serializable. If we make the graph module not serializable due to w_tensor, we will break their flow. I think we can just do key/value pair that somehow references `t_id` in some form.

Reply:
More concretely, I think the format can be in this form:
```
ConstraintSpec = namedtuple(["constraint_dimension", "min", "max"])

# key is tensor id and value is list of constraints for that tensor
input_shape_constraints: Dict[int, List[ConstraintSpec]]
```

Reply:
If you want to avoid tensor ids, another way to index the constraints is by source. I'm doing something related here: https://github.com/pytorch/pytorch/pull/99993 by adding a tensor id to source map in `tx.output.tracked_fakes_id_to_source` which is populated when fake tensors are created.

Since the sources are related to placeholders, I think this would be a stable representation.

Reply:
Serializing t_id is a bad idea anyway, because it's literally the pointer address from the Python process doing the serialization. Want deterministic export artifacts? Kiss them good bye!

Reply:
Will save input constraints as it since the graph module serialization is broken already. A proper fix to make it serializable is needed